src/TLuaInterpreterMapper.cpp

```diff
@@ -3713,15 +3703,12 @@ int TLuaInterpreter::unHighlightRoom(lua_State* L)
     TRoom* pR = host.mpMap->mpRoomDB->getRoom(id);
     if (pR) {
         pR->highlight = false;
-        if (host.mpMap) {
-            if (host.mpMap->mpMapper) {
-                host.mpMap->mpMapper->mp2dMap->update();
-            }
-        }
+        host.mpMap->update();
         lua_pushboolean(L, true);
     } else {
         lua_pushboolean(L, false);
     }
+    host.mpMap->update();
```

Comment:
You've already done this for the case where the room exists - so you don't need to do it again here.

Reply:
Updated.